### ISSUE TITLE
Proxy servers / anti ddos may block AutoModpack 

### ISSUE THREAD

**Issue by N-Montana** [Skidamek/AutoModpack#294]
### Bug description

I'm currently in the process of migrating my server to be a domain instead of direct IP for security issues, however it seems I am unable to have automodpack recognize the domain. No matter what when attempting to connect via the domain it gives an error and the server gives the following:
![image](https://github.com/user-attachments/assets/4c9f8e4a-0304-4522-b5d6-0dbc822b9af2)
The error seems to suggest that using a domain can be done but I'm unable to figure out exactly how to set it up.

### Steps to reproduce

1. Attempt to set automodpack to use a server domain
2. Try to join the server via domain, resulting in the error

### Minecraft & Mod Loader versions

1.18.2 Forge 40.2.21

### Operating system

Unknown (Server is 3rd party hosted)

### AutoModpack version

4.0.0 beta18

### Check list

- [X] I have verified that the issue persists in the latest version of the mod.
- [X] I have searched the existing issues and confirmed that this is not a duplicate.

**Comment by Skidamek** [comment]
Do you use cloudflare dns proxy option? I recently got report that this somehow breaks automodpack host.

**Comment by Skidamek** [comment]
if thats the case please try latest [dev build artifact](https://github.com/Skidamek/AutoModpack/actions/runs/11636422817) its untested may not work but i changed slightly server host handling

**Comment by N-Montana** [comment]
I believe we are (I'm not handling that side personally), I'll test it when I get a chance. Thanks for the quick response!

**Comment by N-Montana** [comment]
Sorry it took a while to test it, it seems to be giving the same issue. I'm assuming that the domain just goes into the "Host IP" config space?

**Comment by Skidamek** [comment]
Yes and localHostIp too. But if empty client will try to get modpack from where it connected to Minecraft server that's why it's optional.

**Comment by N-Montana** [comment]
Okay just wanted to make sure, it does just fail with the same error even when the IP is left blank.

**Comment by Skidamek** [comment]
Well that's unfortunate then, I don't have any other solution for you at the moment however I plan to make major rework of modpack host which should fix thst issue see #293

**Comment by N-Montana** [comment]
Okay that's no problem, we'll make do without for now and try it with the new update. Thanks again for trying help in the meantime!

**Comment by Skidamek** [comment]
cant reproduce it

**Comment by Skidamek** [comment]
or i do, sorry i can
working on fixing it

**Comment by Skidamek** [comment]
could you try new [build](https://github.com/Skidamek/AutoModpack/actions/runs/12073265951) fixes for me

**Comment by N-Montana** [comment]
Annoyingly we're currently in the process of switching to a new modpack and are having a few troubles getting things working properly, once it's up I'll test it!

**Comment by Skidamek** [comment]
Should be fixed in beta 21

**Comment by Skidamek** [comment]
If its not fixed with beta 25, your hosting has to change how they do the port forwarding or just find a different host, i can't do anything about
